client/my-sites/stats/stats-upsell/index.tsx

```diff
@@ -0,0 +1,157 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { isEnabled } from '@automattic/calypso-config';
+import { PLAN_PREMIUM } from '@automattic/calypso-products';
+import page from '@automattic/calypso-router';
+import { Gridicon, PlanPrice } from '@automattic/components';
+import { Plans } from '@automattic/data-stores';
+import formatCurrency from '@automattic/format-currency';
+import { Button } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import TrackComponentView from 'calypso/lib/analytics/track-component-view';
+import useCheckPlanAvailabilityForPurchase from 'calypso/my-sites/plans-features-main/hooks/use-check-plan-availability-for-purchase';
+import { useSelector } from 'calypso/state';
+import { getSiteOption } from 'calypso/state/sites/selectors';
+import { getUpsellModalStatType } from 'calypso/state/stats/paid-stats-upsell/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+
+import './style.scss';
+
+export default function StatsUpsell( { siteId }: { siteId: number } ) {
+	const translate = useTranslate();
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const siteSlug = useSelector( getSelectedSiteSlug );
+	const plans = Plans.usePlans( { coupon: undefined } );
+	const plan = plans?.data?.[ PLAN_PREMIUM ];
+	const pricing = Plans.usePricingMetaForGridPlans( {
+		planSlugs: [ PLAN_PREMIUM ],
+		siteId: selectedSiteId,
+		coupon: undefined,
+		useCheckPlanAvailabilityForPurchase,
+		storageAddOns: null,
+	} )?.[ PLAN_PREMIUM ];
+	const isLoading = plans.isLoading || ! pricing;
+	const isOdysseyStats = isEnabled( 'is_running_in_jetpack_site' );
+	const eventPrefix = isOdysseyStats ? 'jetpack_odyssey' : 'calypso';
+	const isSimpleClassic = useSelector( ( state ) =>
+		getSiteOption( state, selectedSiteId, 'is_wpcom_simple' )
+	);
+	const statType = useSelector( ( state ) => getUpsellModalStatType( state, siteId ) );
+
+	const onClick = ( event: React.MouseEvent< HTMLButtonElement, MouseEvent > ) => {
+		event.preventDefault();
+		recordTracksEvent( `${ eventPrefix }_stats_upsell_submit`, {
+			stat_type: statType,
+		} );
+		if ( isSimpleClassic ) {
```

Comment:
I wonder if we should use `isOdysseyStats` here? `isSimpleClassic` is a subset of that, and I think WoA sites would need the absolute URL as well.

Reply:
hmmm actually let's keep this; theoretically WoA sites wouldn't see this at all. We can come back later and see if we really need to change it.

Reply:
WoA sites can be any plan technically. Leaving for now, can change in follow up if need be.